not compact pg9.6.3

env:
CentOS6
PG9.6.3

operation:

export DATA_SOURCE_NAME="postgresql://postgres:@localhost:5432/xx?sslmode=disable"
./postgres_exporter &
[root@localhost data]# INFO[0000] Established new database connection. source=postgres_exporter.go:951
INFO[0000] Semantic Version Changed: 0.0.0 -> 9.6.3 source=postgres_exporter.go:904
panic: Unknown unit for runtime variable: "16kB"

goroutine 7 [running]:
main.(*pgSetting).metric(0xc4200556d0, 0xc4200a3d98, 0x5)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/pg_setting.go:69 +0x4e0
main.querySettings(0xc42001a960, 0xc4200e5400, 0x0, 0x0)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/pg_setting.go:38 +0x422
main.(*Exporter).scrape(0xc4201100e0, 0xc42001a960)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/postgres_exporter.go:991 +0x4ef
main.(*Exporter).Collect(0xc4201100e0, 0xc42001a960)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/postgres_exporter.go:747 +0x39
main.(*Exporter).Describe(0xc4201100e0, 0xc42001a900)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/postgres_exporter.go:740 +0xb0
github.com/wrouesnel/postgres_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).Register.func1(0x9459c0, 0xc4201100e0, 0xc42001a900)
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:218 +0x3b
created by github.com/wrouesnel/postgres_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).Register
/home/travis/gopath/src/github.com/wrouesnel/postgres_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:220 +0xbc

[1]+ Exit 2 ./postgres_exporter


error with:panic: Unknown unit for runtime variable: "16kB"

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ services:
 - docker
 language: go
 go:
-- '1.8'
+- '1.9'
 # Make sure we have p2 and the postgres client.
 before_install:
 - go get -v github.com/mattn/goveralls

--- a/pg_setting.go
+++ b/pg_setting.go
@@ -94,7 +94,7 @@ func (s *pgSetting) normaliseUnit() (val float64, unit string, err error) {
 		return
 	case "ms", "s", "min", "h", "d":
 		unit = "seconds"
-	case "kB", "MB", "GB", "TB", "8kB", "16MB":
+	case "kB", "MB", "GB", "TB", "8kB", "16kB", "16MB":
 		unit = "bytes"
 	default:
 		err = fmt.Errorf("Unknown unit for runtime variable: %q", s.unit)
@@ -125,6 +125,8 @@ func (s *pgSetting) normaliseUnit() (val float64, unit string, err error) {
 		val *= math.Pow(2, 40)
 	case "8kB":
 		val *= math.Pow(2, 13)
+	case "16kB":
+		val *= math.Pow(2, 14)
 	case "16MB":
 		val *= math.Pow(2, 24)
 	}

--- a/pg_setting_test.go
+++ b/pg_setting_test.go
@@ -62,6 +62,22 @@ var fixtures = []fixture{
 	},
 	{
 		p: pgSetting{
+			name:      "16_kb_real_fixture_metric",
+			setting:   "3.0",
+			unit:      "16kB",
+			shortDesc: "Foo foo foo",
+			vartype:   "real",
+		},
+		n: normalised{
+			val:  49152,
+			unit: "bytes",
+			err:  "",
+		},
+		d: "Desc{fqName: \"pg_settings_16_kb_real_fixture_metric_bytes\", help: \"Foo foo foo [Units converted to bytes.]\", constLabels: {}, variableLabels: []}",
+		v: 49152,
+	},
+	{
+		p: pgSetting{
 			name:      "16_mb_real_fixture_metric",
 			setting:   "3.0",
 			unit:      "16MB",

